Adding Jekyll pages api search

line47 opened the pull request:
Integrating the search with existing functionality. 
https://github.com/18F/jekyll_pages_api_search/

Thanks for the help @afeld 🎉

line47 commented on Gemfile:
```diff
@@ -1,2 +1,6 @@
 gemspec :name => 'uswds-jekyll'
+
+group :jekyll_plugins do
+  gem 'jekyll_pages_api_search'
```
Adding the gem

afeld replied:
You will need to tell users to do this in the README. Adding it to the `gemspec`(s?) is better, but that won't automatically add it to the `jekyll_plugins` `group`.

line47 commented on _config.yml:
```diff
@@ -1 +1,19 @@
 title: This is the site title
+
```
Adding config for search

line47 commented on _data/header.yml:
```diff
@@ -14,9 +14,8 @@ secondary:
-  search: false
+  search:
+    action: /search/
+    query: q
```
changing `param` to `query` as query is what is used in the form code.

line47 commented on _includes/scripts.html:
```diff
@@ -15,3 +15,12 @@
 <script src="{{ _src | replace: _async_marker, '' | relative_url }}"{% if _script.async or _src contains _async_marker %} async{% endif %}></script>
 {% endfor %}{% endfor %}
+
+{% if page.layout == 'search-results' %}
```
Including scripts when the `search-results` layout is used

line47 commented on _layouts/search-results.html:
```diff
@@ -0,0 +1,25 @@
+---
```
Adding search results layout

afeld commented on _includes/scripts.html:
```diff
@@ -15,3 +15,12 @@
+{% if page.layout == 'search-results' %}
+<script>
+var JEKYLL_PAGES_API_SEARCH_UI_OPTIONS = {
+  inputElementId: 'search-field-small'
```
The default uses ID `search-input` - if you change your markup to use that ID instead, you can exclude this `<script>` section.

line47 replied:
Thanks, I've made that update 👍

line47 commented:
Hey @afeld I'm testing this on the https://github.com/18F/frontend guide and I'm close to getting it working I have the search results page loading but I'm not seeing any results.

line47 commented:
@afeld if you can give a second pass at this review that would be great!
I've addressed all the issues I believe and you can see the [frontend-guide](https://github.com/18F/frontend/pull/181) updates as a test.

hursey013 commented:
@line47 I added search in the same way that you've added it here to [another project](https://github.com/GSA/plainlanguage.gov/pull/48/files) and it appears to be working great. The only thing that may improve it is to remove the `*.md` files from a the `assets/uswds/` folder since they are being included in the search results and my attempts to ignore those files via jekyll don't seem to have any effect.

hursey013 commented:
Actually scratch that, it looks like you can ignore that uswds directory using defaults in `_config.yml` https://github.com/GSA/plainlanguage.gov/commit/f69d5ddaf779e8ee2807509b995e150c28116c17

line47 commented:
Thanks @hursey013 I just updated to skip indexing on the `assets/uswds` dir. 🎉 
Was there documentation on the https://github.com/18F/jekyll_pages_api_search repo? If so, I don't see it anywhere.

hursey013 commented:
Yeah I think the only documentation on excluding content with`jekyll_pages_api_search` right now is the bit about being able to use `skip_index: true` on individual pages. Using [defaults](https://jekyllrb.com/docs/configuration/#front-matter-defaults) within the config just allow us to apply that value to entire directories which is more convenient that adding it to each page. In this case, it's not even possible to add it to the markdown files in `assets/uswds` since those are included by the theme gem. I guess you could probably use this approach if you have a site with a lot of content you didn't want indexed and apply `skip_index: true` to everything in the path, and then just whitelist pages you do want included with `skip_index: false`since I _think_ the value in the the actual file will take precedent over the default value.

hursey013 commented on README.md:
```diff
@@ -1,31 +1,37 @@
+---
+skip_index: true
```
I _think_ this may be unnecessary since I don't think this file is ever included into the site that is using the theme gem.
